### ISSUE TITLE
Alerts API returns 400 Error

### ISSUE THREAD

**Issue by SolomonSHY** [googleapis/google-api-python-client#645]
#### Environment details

  - OS: Darwin 18.5.0
  - Python version: 2.7.14
  - pip version: 9.0.3
  - `google-api-python-client` version: 1.7.8

#### Steps to reproduce

  1. Am trying to hit the alerts app, but every endpoint I get has the following error:
`<HttpError 400 when requesting https://alertcenter.googleapis.com/v1beta1/settings?alt=json returned "Request contains an invalid argument.">`
`<HttpError 400 when requesting https://alertcenter.googleapis.com/v1beta1/alerts?alt=json returned "Request contains an invalid argument.">`
etc.

Code: 
```

def create_google_alert_service(keyfile=None, scopes='https://www.googleapis.com/auth/apps.alerts'):
    """Convenience wrapper to create google drive API service."""
    return create_google_api_service('alertcenter', 'v1beta1', scopes, keyfile)

def create_google_api_service(api, version, scopes, keyfile, http=None):
    """
    Create a google api service using service account credentials.
    """
    if keyfile is None:
        raise('Error. You must specify a keyfile.')

    credentials = ServiceAccountCredentials.from_json_keyfile_name(keyfile, scopes)
    return build(api, version, credentials=credentials)

class AlertAPI(object):
    """Alert API object to test the alerts in the system."""

    def __init__(self, page_size=None):
        """Init function for DriveAPI."""
        self.pageSize = page_size or 100
        self.alert_service = create_google_alert_service( secrets.get('google_alerts_keypath'))

    def get_alerts(self):
      alertId = 'XXXX'
      try:
        resp = self.alert_service.v1beta1().getSettings().execute()
        resp = self.alert_service.alerts().feedback().list(alertId=alertId).execute()

        resp = self.alert_service.alerts().get(alertId=alertId).execute()
        resp = self.alert_service.alerts().list().execute()

      except HttpError as e:
        import pdb
        pdb.set_trace()
```

**Comment by grant** [comment]
The error states:
"Request contains an invalid argument"
It seems like the code given has an error.

Currently, we only have samples in Java:
https://developers.google.com/admin-sdk/alertcenter/quickstart/java

**Comment by busunkim96** [comment]
Confirmed with the following (very short) example.

```python
from googleapiclient.discovery import build

client = build("alertcenter", "v1beta1")
resp = client.alerts().list().execute()
```

```
Traceback (most recent call last):
  File "alerts.py", line 7, in <module>
    resp = client.alerts().list().execute()
  File "/usr/local/google/home/busunkim/.local/lib/python3.6/site-packages/googleapiclient/_helpers.py", line 130, in positional_wrapper
    return wrapped(*args, **kwargs)
  File "/usr/local/google/home/busunkim/.local/lib/python3.6/site-packages/googleapiclient/http.py", line 851, in execute
    raise HttpError(resp, content, uri=self.uri)
googleapiclient.errors.HttpError: <HttpError 400 when requesting https://alertcenter.googleapis.com/v1beta1/alerts?alt=json returned "Request contains an invalid argument.">
```

**Comment by busunkim96** [comment]
The error is getting hidden, but it looks like an issue with the customerId.

```
googleapiclient.errors.HttpError: <HttpError 400 when requesting https://alertcenter.googleapis.com/v1beta1/alerts?alt=json returned "Request contains an invalid argument.". Details: "[{'@type': 'type.googleapis.com/google.rpc.DebugInfo', 'detail': "[ORIGINAL ERROR] generic::invalid_argument: com.google.apps.framework.request.BadRequestException: customer_id must be specified.\ncom.google.net.rpc3.client.RpcClientException: <eye3 title='/AppsMicBackendService.ListIncidents, INVALID_ARGUMENT'/> APPLICATION_ERROR;apps.mic/AppsMicBackendService.ListIncidents;com.google.apps.framework.request.BadRequestException: customer_id must be specified.;AppErrorCode=3;StartTimeMs=1565650917893;In process;Deadline(sec)=59.987;ResFormat=AUTOMATIC;Originator=traffic-prod;Tag=\x02cloud_project_number\x03307723894734\x02IncomingMethod\x03/AlertCenterService.ListAlerts\x02cidc\x032;ServerTimeSec=0.01385856;LogBytes=256;Non-FailFast;EffSecLevel=none;ReqFormat=AUTOMATIC;ReqID=7585c3036f7ea230;GlobalID=c53d92865698de92;Server=IN_PROCESS"}]">

```

The documentation says this is an optional parameter, but perhaps this error is raised when the customerId cannot be inferred?

https://developers.google.com/admin-sdk/alertcenter/reference/rest/v1beta1/alerts/list

**Comment by busunkim96** [comment]
(The error field was getting swallowed. That should be resolved with #739)

It looks like normal 3-legged OAuth does not work with this API. There are instructions on how to set up the service account below. Would you recall if you did all the steps below @SolomonSHY?

https://developers.google.com/admin-sdk/alertcenter/guides/prerequisites#set-up-the-alert-center-api


> Set up the Alert Center API
> Follow these steps to set up the Alert Center API:
> 
> 1. Create a service account that can be used by your G Suite application (see instructions on creating a service account).
> 2. Enable the Alert Center API (for instructions, see section on enabling and disabling APIs).
> 3. Grant domain-wide access to the application, and therefore domain-wide delegation of authority to the associated service account (*note that the 3-legged OAuth won't work for the Alert Center API*):
> a. Go to your G Suite domain’s Admin console (see instructions on signing in to your Admin console).
> b. Click Security.
> If you don't see Security listed, select More controls at the bottom of the page, then click Security.
> c. Click Advanced settings.
> d. From the Authentication section, click Manage API client access.
> e. In the Client Name field, type the Client ID for the service account.
> f. In the One or More API Scopes field, enter the list of scopes that your application should be granted access to. In this case, type the following value: https://www.googleapis.com/auth/apps.alerts .
> g. Click Authorize.

**Comment by busunkim96** [comment]
Closing due to lack of activity. Please open a new issue if you are experiencing this problem.
